submodules/common/library.pri

```diff
@@ -0,0 +1,44 @@
+TEMPLATE = lib
+CONFIG -= debug_and_release debug_and_release_target
```

Comment:
Windows builds seem to have this enabled by default. Along with Shadow Builds, which are also enabled by default on Qt5 Windows, the build process will create `/release` and `/debug` directories for any build. These settings will additionally permeate into the generated `*.prl` files, which accompany the compiled libraries and contain importable compilation settings for the BambooTracker subproject. I use these to inherit the detected and explicitly required library dependencies and their required linking flags.

During prl creation, linking flags for Qt5's release libraries along with these two CONFIG settings find their way into the linker settings, even if `CONFIG-=release CONFIG+=debug` is specified. Unsettings these this way fixes the problem. We could use some more sophisticated Qmake logic to instead properly support shadow builds into, reading the correct prl file from and linking against the right vendored Rt\* library from the appropriate `/submodules/Rt{Audio,Midi}/{release,debug}` directory? The current way, if I'm reading the docs correctly, may cause problems when switching between release and debug configurations without `make clean`ing the affected subprojects… But I'm fine with this for now. :slightly_smiling_face:

Reply:
In submodules/common/library.pri:

```qmake
DESTDIR = $$_PRO_FILE_PWD_
OBJECTS_DIR = $$_PRO_FILE_PWD_
MOC_DIR = $$_PRO_FILE_PWD_
```

If TARGET is the same for debug and release, it seems that these settings make qmake generate a prl including both debug and release settings.

While taking advantage of the shadow build, I added "d" suffix to the library names in the case of debug so that the binaries would not be mixed.
RtMidi:
- Dubug: librtmidid.a
- Release: librtmidi.a

[patch.txt](https://github.com/rerrahkr/BambooTracker/files/5742877/patch.txt)

Reply:
Perfect, committed in 34835df!